catalog/nato/stanag4559/nato-stanag4559-source/src/test/java/org/codice/alliance/nato/stanag4559/source/TestStanag4559FilterDelegate.java

```diff
@@ -0,0 +1,830 @@
+
+        List<AttributeInformation> attributeInformationList = generateAttributeInformation().get(
+                Stanag4559Constants.NSIL_ALL_VIEW);
+        StringBuilder result = new StringBuilder(Stanag4559FilterFactory.LP);
+        for (AttributeInformation attributeInformation : attributeInformationList) {
+            if (attributeInformation.attribute_type.equals(AttributeType.TEXT)) {
+                result.append(getPrimary(attributeInformation.attribute_name,
+                        Stanag4559FilterFactory.LIKE,
+                        Stanag4559FilterDelegate.SQ + ATTRIBUTE + Stanag4559FilterDelegate.SQ)
+                        + Stanag4559FilterFactory.OR);
+            }
+
+        }
+        String resultString = result.toString();
+        resultString = resultString.substring(0, result.length() - 4) + Stanag4559FilterFactory.RP;
+        assertThat(filter, is(resultString));
+    }
+
+    @Test
+    public void testPropertyLikeNullQueryableAttributes() {
+        Stanag4559FilterDelegate filterDelegate = new Stanag4559FilterDelegate(null,
+                Stanag4559Constants.NSIL_ALL_VIEW);
+        String filter = filterDelegate.propertyIsLike(ANY_TEXT, ATTRIBUTE, false);
+        assertThat(filter, is(Stanag4559FilterDelegate.EMPTY_STRING));
+    }
+
+    @Test
+    public void testPropertyLikeNullQueryableAttributesForView() {
+        Map<String, List<AttributeInformation>> map = new HashMap<>();
+        map.put(Stanag4559Constants.NSIL_ALL_VIEW, null);
+        Stanag4559FilterDelegate filterDelegate = new Stanag4559FilterDelegate(map,
+                Stanag4559Constants.NSIL_ALL_VIEW);
+        String filter = filterDelegate.propertyIsLike(ANY_TEXT, ATTRIBUTE, false);
+        assertThat(filter, is(Stanag4559FilterDelegate.EMPTY_STRING));
+    }
+
+    @Test
+    public void testPropertyLikeEmptyAttributesForView() {
+        Map<String, List<AttributeInformation>> map = new HashMap<>();
+        List<AttributeInformation> list = new ArrayList<>();
+        map.put(Stanag4559Constants.NSIL_ALL_VIEW, list);
+        Stanag4559FilterDelegate filterDelegate = new Stanag4559FilterDelegate(map,
+                Stanag4559Constants.NSIL_ALL_VIEW);
+        String filter = filterDelegate.propertyIsLike(ANY_TEXT, ATTRIBUTE, false);
+        assertThat(filter, is(Stanag4559FilterDelegate.EMPTY_STRING));
+    }
+
+    @Test
+    public void testAndOneElementList() {
+        List<String> filterList = new ArrayList<>();
+        filterList.add(filterDelegate.propertyIsBetween(PROPERTY, INT, INT));
+        String filter = filterDelegate.and(filterList);
+        assertThat(filter, is(getPrimary(Stanag4559FilterFactory.BTW,
+                INT + Stanag4559FilterFactory.COMMA + INT)));
+    }
+
+    @Test
+    public void testAndEmptyList() {
+        List<String> filterList = new ArrayList<>();
+        String filter = filterDelegate.and(filterList);
+        assertThat(StringUtils.isBlank(filter), is(true));
```

Comment:
If the string has to be empty, there's a matcher you can use: http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/text/IsEmptyString.html#isEmptyString().

But if it's okay for the string to have whitespace, then what you have is fine.